### PR TITLE
Bump to K8s version 22

### PR DESCRIPTION
**Pull request by madalazar**
These are changes needed to make TAS work on K8s version 22, plus some additional refactoring

### CODE REVIEW

**Review by uniemimu — COMMENTED** [review]
lgtm

File: telemetry-aware-scheduling/deploy/extender-configuration/configure-scheduler.sh [hunk 1]
```diff
@@ -1,41 +1,248 @@
...
+scheduler_config_file_path=deploy/extender-configuration/scheduler-config.yaml
+scheduler_config_destination=/etc/kubernetes
 
-## Create the config map and the cluster role for the scheduler configuration. The cluster role is needed in Kubeadm to ensure the scheduler can access configmaps.
-CONFIG_MAP=scheduler-extender-configmap.yaml
-CLUSTER_ROLE=configmap-getter.yaml
+help() {
+  echo "Usage: $(basename "$0") [-m PATH_TO_MANIFEST_FILE] [-f PATH_TO_CONFIGURATION_FILE] [-d CONFIGURATION_DESTINATION_FOLDER] [-th]" 2>&1
+  echo 'Configure the Kubernetes scheduler using one or more of the parameters below. If not entered, the script will be using default values. '
+  echo 'Please make sure the user used to run this script has read & write access to the files/directories mentioned below.'
```
[the remark this answers is not in the document]
Comment (madalazar) [reply telemetry-aware-scheduling/deploy/extender-configuration/configure-scheduler.sh:16]: Done, will be updated in the next commit

File: telemetry-aware-scheduling/deploy/extender-configuration/configure-scheduler.sh [hunk 2]
```diff
@@ -1,41 +1,248 @@
...
+  echo '   -h                                     Show help menu.'
+}
 
+###### PARSE THE CUSTOMER INPUT AND ASSIGN THE VALUES
+while getopts 'm:f:d:th' option; do
+  case $option in
+  h) # print help description
+    help
+    exit;;
+  t) # start the script in test mode. We require parameters to be provided for -m, -f, -d in order to not accidentally
```
[the remark this answers is not in the document]
Comment (madalazar) [reply telemetry-aware-scheduling/deploy/extender-configuration/configure-scheduler.sh:41]: Changed it to "accidentally alter production configuration....". Will be updated in the next commit

**Review by togashidm — APPROVED** [review]
lgtm

**Review by uniemimu — APPROVED** [review]

**Comment by togashidm** [comment]
Thank you for this great work.

**Pull request merged**